syntax = "proto3";
package foo;

message MyMessage {
    int32 num = 1;
    string contents = 2;
}

service Foo {
    rpc sendUnary(MyMessage) returns (MyMessage) {};
    rpc sendStream(stream MyMessage) returns (MyMessage) {};
    rpc sendBiStream(MyMessage) returns (stream MyMessage) {};
}